src/backend/win_cid/file_dialog/dialog_ffi.rs

```diff
@@ -72,6 +72,15 @@ impl IDialog {
     }
 
     fn add_filters(&self, filters: &[crate::dialog::Filter]) -> Result<(), HRESULT> {
+        if let Some(first_filter) = filters.first() {
+            let extension: Vec<u16> = first_filter.extensions[0].encode_utf16().chain(Some(0)).collect();
```

Comment:
`extensions` is specified by the user, which means it can be empty, and this will panic.
I know that providing an empty extensions list is stupid, and it is an obvious user error, but maybe we should use `.get(0)` and handle this case gracefully.

Reply:
Right, I did not realize that the user can add a filter without an extension! Will fix now